code/game/objects/structures/signs/_signs.dm

```diff
@@ -8,6 +10,25 @@
 	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50, "stamina" = 0)
 	var/buildable_sign = 1 //unwrenchable and modifiable
 	rad_flags = RAD_PROTECT_CONTENTS | RAD_NO_CONTAMINATE
+	var/random_dept_base  // will be replaced by a random subtype if the station has random_dept and this is defined
+
+/obj/structure/sign/Initialize(mapload)
+	. = ..()
+	if(random_dept_base)
+		GLOB.dept_signs += src
+
+/obj/structure/sign/proc/randomize()
+	if(!random_dept_base)
+		return
+	var/list/all_dept = subtypesof(random_dept_base)
```

Comment:
Use static since you will use the same value always.

Reply:
so var/static/list/all_dept?

Reply:
Yes